File: src/java/eu/sig/training/ch03/FlagFactoryWithMap.java
package eu.sig.training.ch03;

import static eu.sig.training.ch03.Nationality.BELGIAN;
import static eu.sig.training.ch03.Nationality.DUTCH;
import static eu.sig.training.ch03.Nationality.FRENCH;
import static eu.sig.training.ch03.Nationality.GERMAN;
import static eu.sig.training.ch03.Nationality.ITALIAN;

import java.awt.Color;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class FlagFactoryWithMap {

    // tag::getFlag[]
    private static Map<Nationality, List<Color>> FLAGS =
        new HashMap<Nationality, List<Color>>();

    static {
        FLAGS.put(DUTCH, Arrays.asList(Color.RED, Color.WHITE, Color.BLUE));
        FLAGS.put(GERMAN, Arrays.asList(Color.BLACK, Color.RED, Color.YELLOW));
        FLAGS.put(BELGIAN, Arrays.asList(Color.BLACK, Color.YELLOW, Color.RED));
        FLAGS.put(FRENCH, Arrays.asList(Color.BLUE, Color.WHITE, Color.RED));
        FLAGS.put(ITALIAN, Arrays.asList(Color.GREEN, Color.WHITE, Color.RED));
    }

    public List<Color> getFlagColors(Nationality nationality) {
        List<Color> colors = FLAGS.get(nationality);
        return colors != null ? colors : Arrays.asList(Color.GRAY);
    }
    // end::getFlag[]

}

File: src/csharp/eu/sig/training/ch03/FlagFactoryWithMap.cs
using System.Drawing;
using System.Collections.Generic;

namespace eu.sig.training.ch03
{

    public class FlagFactoryWithMap
    {

        // tag::getFlag[]
        private static Dictionary<Nationality, IList<Color>> FLAGS =
            new Dictionary<Nationality, IList<Color>>();

        static FlagFactoryWithMap()
        {
            FLAGS[Nationality.DUTCH] = new List<Color>{ Color.Red, Color.White,
                Color.Blue };
            FLAGS[Nationality.GERMAN] = new List<Color>{ Color.Black, Color.Red,
                Color.Yellow };
            FLAGS[Nationality.BELGIAN] = new List<Color>{ Color.Black, Color.Yellow,
                Color.Red };
            FLAGS[Nationality.FRENCH] = new List<Color>{ Color.Blue, Color.White,
                Color.Red };
            FLAGS[Nationality.ITALIAN] = new List<Color>{ Color.Green, Color.White,
                Color.Red };
        }

        public IList<Color> GetFlagColors(Nationality nationality)
        {
            IList<Color> colors = FLAGS[nationality];
            return colors ?? new List<Color> { Color.Gray };
        }
        // end::getFlag[]
    }
}
